pkg/cmd/secret/list/list.go

```diff
@@ -0,0 +1,156 @@
+package list
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghinstance"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/secret/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type ListOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	OrgName string
+}
+
+func NewCmdList(f *cmdutil.Factory, runF func(*ListOptions) error) *cobra.Command {
+	opts := &ListOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List secrets",
+		Long:  "List secrets for a repository or organization",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return listRun(opts)
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.OrgName, "org", "", "List secrets for an organization")
+	cmd.Flags().Lookup("org").NoOptDefVal = "@owner"
```

Comment:
For org secrets, would we consider requiring that you always specify the full org name, instead of having a NoOptDefVal flag here?

```
gh secret list --org github
gh secret set MY_SEKRIT --org github
```

This would be to avoid having to require an `=` sign for specifying a value, which might be surprising to our users.

Reply:
i'm fine with that